### PR TITLE
Issue/2033 Made Search API always Match All Input  keywords

### LINKED ISSUE #2033: Partially matched result may look counter-intuitive & Probably need to make sure Filter options are non-negotiable
### Problem description

The following links:
https://data.gov.au/search?format=WMS&organisation=CSIRO%20Oceans%20and%20Atmosphere&q=%2A
shows that:

There are `43` documents match `CSIRO Oceans and Atmosphere`

image

But `5832` documents match `WMS`

image

And `10327` results are returned

image

Those different figures could confuse users, although it is correct behaviour --- when first search attempt can't find result we will ignore some of the filters.

### Proposed Solutions:

- We probably should make sure filter options are never ignored. 
- If we want to offer more results to the user as suggestion, we probably should generate alternative query by altering the query string only (i.e. never cancel or modify filter options).

One example could be what google does:

image

Or what ebay does:

image

- If those options are technically hard in short term, we should consider remove the partially match feature

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Fixes #2033 
Also fixed: #2025 & #2065 

Change Summary:

- Removed partial match logic
- Pull location keywords from input text and query as all keywords OR non location keywords plus geoQuery
- Default Operator should be AND for distribution query
- Fixed: LanguageAnalyzerSpec.scala
  - Only randomly pick one distribution to test as now it's AND operator in simple_string_query. All keywords must exist in one distribution title
- Adjusted RegionName Boost test cases logic 
- Ignore inexact test cases
- Added mapping for accrualPeriodicity
- corrected not exist field name `contactPoint.name` -> `contactPoint.identifier`
- moved `NON_LANGUAGE_FIELDS` to a seperate simple_string_query so they won't generated always false query terms

### Test Site:

https://issue-2033.dev.magda.io

### Checklist

-   [x] There are unit tests to verify my changes are correct
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

File: magda-int-test/src/test/scala/au/csiro/data61/magda/api/DataSetSearchSpec.scala [hunk 1]
```diff
@@ -641,7 +669,7 @@ class DataSetSearchSpec extends BaseSearchApiSpec with RegistryConverters {
         }
       }
 
-      it("inexact") {
+      ignore("inexact") {
```
Comment (AlexGilleran) [inline magda-int-test/src/test/scala/au/csiro/data61/magda/api/DataSetSearchSpec.scala]: I think we should just delete these if we don't need them, can always get them out of git later if we need them.

File: .travis.yml [hunk 2]
```diff
@@ -1,75 +1,75 @@
 dist: trusty
 sudo: false
 matrix:
-  include:
```
Comment (AlexGilleran) [inline .travis.yml]: Looks like a lot of these yml files have had 4 space formatting applied, can we revert that? If it's prettier's fault we might have to put new rules in prettier rc.

File: magda-int-test/src/test/scala/au/csiro/data61/magda/api/DataSetSearchSpec.scala (hunk 1 above)
Comment (t83714) [reply magda-int-test/src/test/scala/au/csiro/data61/magda/api/DataSetSearchSpec.scala]: Deleted~ 👍

File: .travis.yml (hunk 2 above)
Comment (t83714) [reply .travis.yml]: Yeah~ it's triggered by prettier pre-commit hook~
Overrider it to 2 spaces indent.
It seems pettier enforce an `indentation style for sequence in mapping`. 
i.e. 
```yaml
paths:
- .yarn
- "yarn.lock"
```
Will become:
```yaml
paths:
  - .yarn
  - "yarn.lock"
```
See:
https://github.com/prettier/prettier/issues/4723
If that's not desired style, we probably have to add those yaml files to ignore list.

**Comment by AlexGilleran** [comment]
Just realised this also needs the index version bumped in `common.conf` 😬

File: .travis.yml (hunk 2 above)
Comment (AlexGilleran) [reply .travis.yml]: Ah right. Yeah I suppose the new style is neater, makes for an ugly diff though 😬

**Comment by t83714** [comment]
> Just realised this also needs the index version bumped in `common.conf` 😬

Thanks a lot for this 👍 Have added a new commit~

**Pull request merged**